app/models/loan.rb

```diff
@@ -234,4 +234,24 @@ def project_events(order_by="Completed IS NULL OR Completed = '0000-00-00', Comp
   def logs(order_by="Date DESC")
     @logs ||= ProjectLog.where("lower(ProjectTable) = 'loans' and ProjectID = ?", self.ID).order(order_by)
   end
+
+  # The below methods may need to be moved to the events model
+    def calendar_start_event
+      cal_item = {}
+      cal_item[:start] = self.signing_date
+      cal_item[:title] = "Start " + self.name
+      cal_item[:event_type] = "loan_start"
+
+      return cal_item
+    end
+
+    def calendar_end_event
+      cal_item = {}
+      cal_item[:start] = self.target_end_date
+      cal_item[:title] = "End " + self.name
+      cal_item[:event_type] = "loan_end"
```

Comment:
This is a great first take on building these objects. Let's talk about a slightly more robust way of doing so. I realize this is model level code and not your speciality, and you've done a great first pass here!